common/txmgr/types/tx.go

```diff
@@ -66,6 +66,7 @@ func (s TxAttemptState) String() (str string) {
 }
 
 type TxRequest[ADDR types.Hashable, TX_HASH types.Hashable] struct {
+	RequestID        *TX_HASH
```

Comment:
Since you are adding this field now, can you please add code comments indicating what this field means, nd how is this used by the TXM?